### DISCUSSION TITLE
Unable to raise exception after checking a specific condition

### DISCUSSION THREAD

**Discussion by debuggerrr** [cue-lang/cue#2224] [Q&A]
I want to raise an issue for suppose below example:  
`
if len(a)>0 {
raise exception
}
`

How can be it done using CUE? I didn't find any specific example w.r.t raising an exception.

**Comment by slewiskelly** [comment]
I suppose you could do something like:

```cue
if len(a) > 0 {
	_err: _|_
}
```

But it wouldn't be an encouraged pattern. 

I think maybe you are trying employ concepts and techniques from imperative, general-purpose, programming languages that don't apply to CUE.

In CUE, you would add a constrain to specify that `a` should contain (at least, exactly, or no more than) a specifc number of elements.

Using your specific example, you could use either of the following:

```cue
a: []
```

```cue
import "list"

a: [...string] & list.MaxItems(0)
```

```cue
a: [...string]

_ok: true & len(a) < 1
```

Choosing one over the other would depend on a specific situation, but the first two are likely suitable for the majority of cases.

**Reply by debuggerrr** [reply]
Okay, let me try this tomorrow. Thanks for the help though :)

**Comment by myitcv** [comment]
@debuggerrr can you provide a bit more information about what you're trying to do here? 

How/where are you using CUE?

In a policy setting?
